lib/ViroDB/ResultSet/PrimerSearch.pm

```diff
@@ -0,0 +1,56 @@
+use strict;
+use warnings;
+use 5.018;
+use utf8;
+
+package ViroDB::ResultSet::PrimerSearch;
+use Moose;
+use MooseX::NonMoose;
+use MooseX::MarkAsMethods autoclean => 1;
+extends 'ViroDB::ResultSet';
+
+with 'ViroDB::Helper::ResultSet::SearchArrayOverlaps';
+with 'ViroDB::Helper::ResultSet::SearchFreeform', {
+    text_fields => [qw[
+        name
+        sequence
+        orientation
+        notes
+        date_added
+        organism
+        positions
+    ]],
+    id_field    => "primer_id",
+};
+
+sub plausible_for {
+    my ($self, $query) = @_;
+    my $me    = $self->current_source_alias;
+    my $where = qq{
+                    regexp_replace(upper($me.name), '[^A-Z0-9]', '', 'g')
+        LIKE '%' || regexp_replace(upper(?),        '[^A-Z0-9]', '', 'g') || '%'
+    };
+
+    return $self->search(\[ $where, $query ])
+        ->order_by(\[ "length(?)::float / length($me.name) desc, $me.name asc", $query ]);
+}
+
+sub organism {
+    my $self = shift;
+    my $me   = $self->current_source_alias;
+    return $self->search({ "$me.organism" => \@_ });
+}
+
+sub orientation {
+    my $self = shift;
+    my $me   = $self->current_source_alias;
+    return $self->search({ "$me.orientation" => \@_ });
+}
+
+sub position {
+    my $self = shift;
+    return $self->search_array_overlaps( positions => @_ );
```

Comment:
When looking for an existing primer that meets one's needs, an exact position match isn't necessarily required. If searching by numerical reference position, finding all _nearby_ primers would be more useful. (Probably simplest to hardcode a max difference (15? 30?) for starters.)

Reply:
I have no idea how I'd go about implementing that.

Reply:
Hm it's probably not quite as straightforward to implement as I was imagining because of the array column, go ahead and punt this to a new card.